microceph/api/remote.go

```diff
@@ -0,0 +1,166 @@
+package api
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/lxd/shared/logger"
+	"github.com/canonical/microceph/microceph/api/types"
+	"github.com/canonical/microceph/microceph/ceph"
+	"github.com/canonical/microceph/microceph/client"
+	"github.com/canonical/microceph/microceph/constants"
+	"github.com/canonical/microceph/microceph/database"
+	"github.com/canonical/microceph/microceph/interfaces"
+	"github.com/canonical/microcluster/rest"
+	"github.com/canonical/microcluster/state"
+	"github.com/gorilla/mux"
+)
+
+var remoteCmd = rest.Endpoint{
+	Path: "client/remotes",
+	Get:  rest.EndpointAction{Handler: CmdRemoteGet, ProxyTarget: false},
+}
+
+var remoteNameCmd = rest.Endpoint{
+	Path:   "client/remotes/{name}",
+	Put:    rest.EndpointAction{Handler: CmdRemotePut, ProxyTarget: false},
+	Get:    rest.EndpointAction{Handler: CmdRemoteGet, ProxyTarget: false},
+	Delete: rest.EndpointAction{Handler: CmdRemoteDelete, ProxyTarget: false},
+}
+
+var CmdRemotePut = func(state *state.State, r *http.Request) response.Response {
```

Comment:
Question, we typically don't make the handler funcs public -- is there a reason why this handler var and the ones below are pub?

Reply:
public function variable can be stubbed with mock implementations for unit testing. I have consistently added functions in this manner but IG API handlers won't need it. I will make them private.